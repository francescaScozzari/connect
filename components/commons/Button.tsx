import styled from 'styled-components'

export const Button = styled.button`
  align-items: center;
  background-color: white;
  border: 2px solid black;
  border-radius: 10px;
  cursor: pointer;
  display: flex;
  justify-content: center;
  min-width: 120px;
  outline: none;
  padding: 1em 1.5em;
  gap: 1em;
  transition: 0.3s background-color ease-in-out;
  width: fit-content;

  &:disabled {
    cursor: not-allowed;
    opacity: 0.4;
  }
`
