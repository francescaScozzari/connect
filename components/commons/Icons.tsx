import React from 'react'

type Props = {
  title: string
} & React.ComponentPropsWithoutRef<'svg'>

const IconArrowCircle = ({ title, ...props }: Props) => {
  return (
    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24" {...props}>
      <title>{title}</title>
      <path d="M12 1a11 11 0 1 0 11 11A11 11 0 0 0 12 1Zm4.707 9.707a1 1 0 0 1-1.414 0L13 8.414V18a1 1 0 0 1-2 0V8.414l-2.293 2.293a1 1 0 1 1-1.414-1.414l4-4a1 1 0 0 1 1.414 0l4 4a1 1 0 0 1 0 1.414Z" />
    </svg>
  )
}

const IconArrowSubmit = ({ title, ...props }: Props) => {
  return (
    <svg
      width="45"
      height="45"
      viewBox="0 0 45 45"
      fill="none"
      xmlns="http://www.w3.org/2000/svg"
      {...props}
    >
      <title>{title}</title>
      <g id="Arrow forward">
        <path
          id="Vector"
          d="M22.5 7.5L19.8563 10.1437L30.3188 20.625H7.5V24.375H30.3188L19.8563 34.8563L22.5 37.5L37.5 22.5L22.5 7.5Z"
          fill="black"
        />
      </g>
    </svg>
  )
}

const IconCrossReset = ({ title, ...props }: Props) => {
  return (
    <svg
      width="31"
      height="31"
      fill="none"
      xmlns="http://www.w3.org/2000/svg"
      {...props}
    >
      <title>{title}</title>
      <g>
        <path
          d="m24.5417 8.27962-1.8212-1.82125L15.5 13.6788 8.27962 6.45837 6.45837 8.27962 13.6788 15.5l-7.22043 7.2205 1.82125 1.8212L15.5 17.3213l7.2205 7.2204 1.8212-1.8212L17.3213 15.5l7.2204-7.22038Z"
          fill="#000"
        />
      </g>
    </svg>
  )
}

const IconCopy = ({ title, ...props }: Props) => {
  return (
    <svg
      width="49"
      height="49"
      fill="none"
      xmlns="http://www.w3.org/2000/svg"
      {...props}
    >
      <title>{title}</title>
      <g>
        <path
          d="M32.6673 2.04163H8.16732c-2.24584 0-4.08334 1.8375-4.08334 4.08333V34.7083h4.08334V6.12496H32.6673V2.04163Zm-2.0416 8.16667H16.334c-2.2458 0-4.0629 1.8375-4.0629 4.0833l-.0204 28.5834c0 2.2458 1.817 4.0833 4.0629 4.0833h22.4787c2.2459 0 4.0834-1.8375 4.0834-4.0833V22.4583l-12.25-12.25ZM16.334 42.875V14.2916h12.25V24.5h10.2083v18.375H16.334Z"
          fill="#000"
        />
      </g>
    </svg>
  )
}

const IconTarget = ({ title, ...props }: Props) => {
  return (
    <svg
      width="49"
      height="51"
      fill="none"
      xmlns="http://www.w3.org/2000/svg"
      {...props}
    >
      <title>{title}</title>
      <path
        d="M24.4993 16.9969c-4.512 0-8.1666 3.803-8.1666 8.4984s3.6546 8.4985 8.1666 8.4985c4.5121 0 8.1667-3.8031 8.1667-8.4985 0-4.6954-3.6546-8.4984-8.1667-8.4984Zm18.2525 6.3738c-.9391-8.8596-7.697-15.89205-16.2108-16.86937V4.24924c0-1.16853-.9187-2.12461-2.0417-2.12461-1.1229 0-2.0416.95608-2.0416 2.12461v2.25209C13.9439 7.47865 7.18602 14.5111 6.24685 23.3707H4.08268c-1.12291 0-2.04166.9561-2.04166 2.1246 0 1.1686.91875 2.1246 2.04166 2.1246h2.16417c.93917 8.8597 7.69705 15.8921 16.21085 16.8694v2.2521c0 1.1686.9187 2.1246 2.0416 2.1246 1.123 0 2.0417-.956 2.0417-2.1246v-2.2521c8.5138-.9773 15.2717-8.0097 16.2108-16.8694h2.1642c1.1229 0 2.0417-.956 2.0417-2.1246 0-1.1685-.9188-2.1246-2.0417-2.1246h-2.1642ZM24.4993 40.3676c-7.9012 0-14.2916-6.65-14.2916-14.8723 0-8.2222 6.3904-14.8722 14.2916-14.8722 7.9013 0 14.2917 6.65 14.2917 14.8722 0 8.2223-6.3904 14.8723-14.2917 14.8723Z"
        fill="#000"
      />
    </svg>
  )
}

export {
  IconArrowCircle,
  IconArrowSubmit,
  IconCrossReset,
  IconCopy,
  IconTarget
}
