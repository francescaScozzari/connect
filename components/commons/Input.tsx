import React from 'react'
import styled from 'styled-components'

type Props = JSX.IntrinsicElements['input'] & {
  errors?: string[]
}

const InputWithErrors = ({
  errors,
  max,
  min,
  onChange,
  placeholder,
  required,
  step,
  style,
  type,
  value
}: Props) => {
  return (
    <InputContainer style={style}>
      <Input
        errors={errors && errors.length > 0}
        max={max}
        min={min}
        onChange={onChange}
        placeholder={placeholder}
        required={required}
        step={step}
        type={type}
        value={value}
      />
      {errors && errors.length > 0 && errors.map(err => <p key={err}>{err}</p>)}
    </InputContainer>
  )
}

const Input = styled.input<{ errors?: boolean }>`
  border: 2px solid black;
  box-sizing: border-box;
  border-radius: 0.5em;
  padding: 0.625em 2em;
  width: 25em;
  font-style: normal;
  font-size: 1.125rem;
  &::placeholder {
    padding-left: 1.5em;
    font-size: 1.125rem;
    background: url("data:image/svg+xml,%3Csvg height='24' width='24' xmlns='http://www.w3.org/2000/svg'%3E%3Cdefs%3E%3Cstyle%3E.cls-1%7Bfill:none;stroke:%23000;stroke-linecap:round;stroke-linejoin:round;stroke-width:2px%7D%3C/style%3E%3C/defs%3E%3Cg id='_21.search'%3E%3Ccircle class='cls-1' cx='9' cy='9' r='8'/%3E%3Cpath class='cls-1' d='m15 15 8 8'/%3E%3C/g%3E%3C/svg%3E")
      no-repeat 0.5em center;
  }
  @media (max-width: 768px) {
    font-size: 1em;
    width: 15em;
  }
`

const InputContainer = styled.div`
  display: flex;
  flex-direction: column;
  margin-bottom: 1.5em;
  p {
    font-size: 0.75em;
    margin-left: 0.625em;
    color: ${({ theme }) => theme.colors.status.error};
  }
`

export { Input, InputWithErrors }
