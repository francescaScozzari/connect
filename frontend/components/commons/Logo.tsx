import React from 'react'

type Props = {
  title: string
} & React.ComponentPropsWithoutRef<'svg'>

const BigLogo = ({ title, ...props }: Props) => {
  return (
    <svg
      width="464"
      height="144"
      fill="none"
      xmlns="http://www.w3.org/2000/svg"
      {...props}
    >
      <title>{title}</title>
      <path
        d="M66.76 6.354H13.746V62.27C20.298 32.603 37.763 13.914 66.76 6.354Z"
        fill="#009ADE"
      />
      <path
        d="M13.743 83.162v55.183h54.184c-29.667-7.208-47.564-24.665-54.184-55.183Z"
        fill="#ABC511"
      />
      <path
        d="M79.406 124.382c-2.487-13.75-8.015-25.162-17.364-34.468-9.316-9.277-20.58-14.906-34.166-17.172 28.477-6.246 45.495-23.25 51.759-52.135 2.235 13.98 7.97 25.276 17.289 34.652 9.319 9.375 20.667 14.882 34.182 17.25-28.372 6.079-45.331 22.935-51.7 51.873Z"
        fill="#fff"
      />
      <path
        d="M145.078 82.62v55.725H90.973c29.666-7.208 47.481-25.207 54.105-55.725Z"
        fill="#DF057E"
      />
      <path
        d="M92.055 6.354h53.023V62.05c-6.555-29.667-24.03-48.136-53.023-55.696Z"
        fill="#0C927A"
      />
      <path
        d="M171.017 135.928c0 1.79-1.183 2.811-3.549 2.811h-4.29v-10.63h4.016c2.061 0 3.316.956 3.316 2.719 0 1.261-.647 2.082-1.611 2.43 1.363.286 2.118 1.199 2.118 2.673v-.003Zm-7.137-7.152v4.233h3.284c1.663 0 2.627-.713 2.627-2.125s-.964-2.108-2.627-2.108h-3.284Zm6.434 7.09c0-1.517-1.003-2.187-2.846-2.187h-3.588v4.387h3.588c1.843 0 2.846-.67 2.846-2.2ZM172.769 133.422c0-3.083 2.13-5.392 5.006-5.392 2.875 0 5.006 2.292 5.006 5.392 0 3.101-2.147 5.389-5.006 5.389-2.86 0-5.006-2.308-5.006-5.389Zm9.296 0c0-2.689-1.843-4.679-4.29-4.679-2.448 0-4.304 1.99-4.304 4.679 0 2.69 1.843 4.677 4.304 4.677 2.46 0 4.29-1.987 4.29-4.677ZM184.535 133.422c0-3.083 2.131-5.392 5.006-5.392 2.876 0 5.006 2.292 5.006 5.392 0 3.101-2.147 5.389-5.006 5.389-2.859 0-5.006-2.308-5.006-5.389Zm9.296 0c0-2.689-1.842-4.679-4.29-4.679-2.447 0-4.303 1.99-4.303 4.679 0 2.69 1.843 4.677 4.303 4.677 2.461 0 4.29-1.987 4.29-4.677ZM196.112 137.37l.317-.575c.66.759 1.886 1.333 3.176 1.333 1.928 0 2.778-.926 2.778-2.078 0-3.189-6.009-1.278-6.009-5.192 0-1.501 1.045-2.824 3.408-2.824 1.059 0 2.16.364 2.901.955l-.261.624a4.467 4.467 0 0 0-2.64-.896c-1.869 0-2.709.955-2.709 2.124 0 3.189 6.009 1.307 6.009 5.163 0 1.504-1.085 2.811-3.464 2.811-1.431 0-2.803-.591-3.506-1.442v-.003ZM207.706 128.805h-3.464v-.699h7.633v.699h-3.467v9.931h-.699v-9.931h-.003ZM213.74 128.106h.703v10.63h-.703v-10.63ZM225.576 128.106v10.63h-.578l-6.558-9.235v9.235h-.703v-10.63h.592l6.545 9.234v-9.234h.702ZM235.911 133.465h.677v3.977c-.866.91-2.118 1.366-3.438 1.366-2.901 0-5.032-2.292-5.032-5.389 0-3.097 2.131-5.392 5.049-5.392 1.388 0 2.61.486 3.463 1.458l-.441.516c-.84-.91-1.843-1.261-3.013-1.261-2.503 0-4.359 2.006-4.359 4.679 0 2.673 1.856 4.676 4.359 4.676 1.017 0 1.951-.272 2.735-.968v-3.662ZM243.077 128.106h.703v10.63h-.703v-10.63ZM254.912 128.106v10.63h-.578l-6.558-9.235v9.235h-.703v-10.63h.592l6.545 9.234v-9.234h.702ZM265.251 133.465h.677v3.977c-.866.91-2.118 1.366-3.438 1.366-2.902 0-5.032-2.292-5.032-5.389 0-3.097 2.13-5.392 5.048-5.392 1.389 0 2.611.486 3.464 1.458l-.441.516c-.84-.91-1.843-1.261-3.013-1.261-2.503 0-4.359 2.006-4.359 4.679 0 2.673 1.856 4.676 4.359 4.676 1.017 0 1.951-.272 2.735-.968v-3.662ZM275.613 138.036v.7h-6.643v-10.63h6.437v.699h-5.735v4.191h5.13v.683h-5.13v4.357h5.941ZM285.769 128.106v10.63h-.579l-6.558-9.235v9.235h-.702v-10.63h.591l6.545 9.234v-9.234h.703ZM289.078 128.106h.703v10.63h-.703v-10.63ZM293.016 134.224v-6.118h.699v6.088c0 2.657 1.157 3.901 3.163 3.901 2.007 0 3.15-1.244 3.15-3.901v-6.088h.7v6.118c0 3.037-1.471 4.584-3.866 4.584s-3.849-1.547-3.849-4.584h.003ZM313.076 138.736l-.013-9.097-4.029 7.685h-.343l-4.029-7.655v9.067h-.686v-10.63h.591l4.303 8.213 4.304-8.213h.578l.017 10.63H313.076ZM321.229 128.805v4.539h5.13v.699h-5.13v4.693h-.703v-10.63h6.438v.699h-5.735ZM328.329 133.422c0-3.083 2.131-5.392 5.006-5.392 2.876 0 5.006 2.292 5.006 5.392 0 3.101-2.147 5.389-5.006 5.389-2.859 0-5.006-2.308-5.006-5.389Zm9.296 0c0-2.689-1.842-4.679-4.29-4.679-2.447 0-4.303 1.99-4.303 4.679 0 2.69 1.843 4.677 4.303 4.677 2.461 0 4.29-1.987 4.29-4.677ZM347.569 138.736l-2.297-3.553a6.126 6.126 0 0 1-.935.075h-2.748v3.478h-.702v-10.63h3.45c2.422 0 3.879 1.336 3.879 3.583 0 1.684-.827 2.853-2.255 3.326l2.405 3.721h-.797Zm-3.232-4.161c2.062 0 3.163-1.064 3.163-2.883s-1.101-2.887-3.163-2.887h-2.748v5.77h2.748ZM360.838 138.036v.7h-6.643v-10.63h6.438v.699h-5.735v4.191h5.13v.683h-5.13v4.357h5.94ZM369.837 138.736l-3.313-4.919-3.3 4.919h-.824l3.699-5.481-3.437-5.149h.823l3.069 4.571 3.065-4.571h.784l-3.438 5.133 3.699 5.497h-.827ZM371.511 133.422c0-3.1 2.13-5.392 4.993-5.392 1.359 0 2.584.486 3.424 1.475l-.441.499c-.823-.897-1.83-1.261-2.957-1.261-2.464 0-4.32 2.006-4.32 4.679 0 2.674 1.856 4.677 4.32 4.677 1.127 0 2.13-.378 2.957-1.274l.441.502c-.84.985-2.062 1.488-3.424 1.488-2.86 0-4.993-2.293-4.993-5.389v-.004ZM388.881 138.036v.7h-6.643v-10.63h6.437v.699h-5.734v4.191h5.13v.683h-5.13v4.357h5.94ZM391.201 128.106h.703v9.93h5.542v.7h-6.245v-10.63ZM399.354 128.106h.702v9.93h5.542v.7h-6.244v-10.63ZM414.15 138.036v.7h-6.643v-10.63h6.437v.699h-5.735v4.191h5.131v.683h-5.131v4.357h5.941ZM424.309 128.106v10.63h-.579l-6.558-9.235v9.235h-.702v-10.63h.591l6.545 9.234v-9.234h.703ZM426.848 133.422c0-3.1 2.13-5.392 4.993-5.392 1.359 0 2.584.486 3.424 1.475l-.441.499c-.824-.897-1.83-1.261-2.957-1.261-2.464 0-4.32 2.006-4.32 4.679 0 2.674 1.856 4.677 4.32 4.677 1.127 0 2.13-.378 2.957-1.274l.441.502c-.84.985-2.062 1.488-3.424 1.488-2.86 0-4.993-2.293-4.993-5.389v-.004ZM444.218 138.036v.7h-6.643v-10.63h6.437v.699h-5.734v4.191h5.13v.683h-5.13v4.357h5.94ZM227.036 58.175c9.388 4.916 15.28 14.758 15.28 26.767 0 17.829-13.93 31.332-32.32 31.332h-45.92V6.246h45.002c17.664 0 30.555 11.904 30.555 28.16 0 9.832-4.986 18.675-12.597 23.769ZM213.58 39.427c0-5.71-3.542-9.58-9.283-9.58h-13.871v19.16h13.871c5.741 0 9.283-3.87 9.283-9.58Zm1.585 41.913c0-5.842-3.885-9.96-9.829-9.96h-14.91v20.008h14.91c5.944 0 9.829-4.114 9.829-10.052v.004ZM254.945 6.243h27.811V116.27h-27.811V6.243ZM368.687 97.493h-8.551v18.781h-24.321v-18.78h-45.436V92.38l30.32-80.637h25.134L322.392 74.07h15.135l5.552-23.02h17.053v23.02h8.552v23.42l.003.004ZM444.218 89.98v26.291h-64.953V6.243h63.747v26.202h-36.42v16.197h32.721v24.862h-32.721v16.472h37.626v.004Z"
        fill="#fff"
      />
    </svg>
  )
}

const SmallLogo = ({ title, ...props }: Props) => {
  return (
    <svg
      width="202"
      height="61"
      fill="none"
      xmlns="http://www.w3.org/2000/svg"
      {...props}
    >
      <title>{title}</title>
      <path
        d="M29.064 2.692H5.984v23.686C8.838 13.811 16.44 5.894 29.065 2.692Z"
        fill="#009ADE"
      />
      <path
        d="M5.983 35.228v23.377h23.589C16.657 55.55 8.865 48.154 5.983 35.228Z"
        fill="#ABC511"
      />
      <path
        d="M34.57 52.69c-1.083-5.825-3.49-10.66-7.56-14.601-4.056-3.93-8.96-6.315-14.874-7.275C24.533 28.17 31.942 20.965 34.669 8.73c.973 5.922 3.47 10.708 7.526 14.68 4.057 3.97 8.998 6.304 14.882 7.307C44.725 33.29 37.342 40.43 34.569 52.69Z"
        fill="#fff"
      />
      <path
        d="M63.16 34.999v23.605H39.604C52.52 55.552 60.275 47.927 63.159 35Z"
        fill="#DF057E"
      />
      <path
        d="M40.076 2.692h23.083v23.593C60.306 13.718 52.698 5.895 40.076 2.692Z"
        fill="#0C927A"
      />
      <path
        d="M74.451 57.58c0 .759-.515 1.192-1.545 1.192H71.04v-4.503h1.748c.897 0 1.444.404 1.444 1.151 0 .535-.282.882-.702 1.03.594.12.922.508.922 1.132v-.001Zm-3.107-3.03v1.794h1.43c.724 0 1.144-.302 1.144-.9 0-.598-.42-.893-1.144-.893h-1.43Zm2.801 3.004c0-.642-.436-.926-1.239-.926h-1.562v1.858h1.562c.803 0 1.24-.283 1.24-.932ZM75.214 56.52c0-1.307.927-2.285 2.18-2.285 1.251 0 2.178.971 2.178 2.284 0 1.314-.934 2.283-2.179 2.283-1.245 0-2.18-.978-2.18-2.283Zm4.047 0c0-1.14-.802-1.983-1.868-1.983-1.065 0-1.873.843-1.873 1.982 0 1.14.802 1.981 1.873 1.981s1.868-.841 1.868-1.98ZM80.336 56.52c0-1.307.928-2.285 2.18-2.285 1.252 0 2.18.971 2.18 2.284 0 1.314-.936 2.283-2.18 2.283-1.245 0-2.18-.978-2.18-2.283Zm4.047 0c0-1.14-.802-1.983-1.867-1.983-1.066 0-1.874.843-1.874 1.982 0 1.14.803 1.981 1.874 1.981 1.07 0 1.867-.841 1.867-1.98ZM85.376 58.191l.138-.243c.288.321.821.565 1.383.565.84 0 1.21-.392 1.21-.88 0-1.352-2.617-.542-2.617-2.2 0-.636.456-1.196 1.484-1.196.46 0 .94.154 1.263.404l-.114.265a1.98 1.98 0 0 0-1.149-.38c-.814 0-1.18.405-1.18.9 0 1.35 2.617.554 2.617 2.187 0 .637-.472 1.19-1.508 1.19-.623 0-1.22-.25-1.527-.61v-.002ZM90.423 54.563h-1.508v-.296h3.324v.296h-1.51v4.207h-.304v-4.207h-.002ZM93.05 54.267h.307v4.503h-.306v-4.503ZM98.203 54.267v4.503h-.252l-2.855-3.912v3.912h-.306v-4.503h.258l2.85 3.912v-3.912h.305ZM102.703 56.537h.294v1.685c-.377.385-.922.579-1.496.579-1.263 0-2.19-.971-2.19-2.283 0-1.312.927-2.284 2.197-2.284.604 0 1.136.206 1.508.617l-.192.219c-.366-.386-.803-.535-1.312-.535-1.09 0-1.897.85-1.897 1.983 0 1.132.807 1.98 1.897 1.98.443 0 .849-.115 1.191-.41v-1.55ZM105.822 54.267h.306v4.503h-.306v-4.503ZM110.975 54.267v4.503h-.252l-2.855-3.912v3.912h-.306v-4.503h.257l2.85 3.912v-3.912h.306ZM115.476 56.537h.294v1.685c-.377.385-.922.579-1.496.579-1.263 0-2.191-.971-2.191-2.283 0-1.312.928-2.284 2.198-2.284.604 0 1.136.206 1.508.617l-.192.219c-.366-.386-.803-.535-1.312-.535-1.09 0-1.898.85-1.898 1.983 0 1.132.808 1.98 1.898 1.98.443 0 .849-.115 1.191-.41v-1.55ZM119.987 58.474v.296h-2.892v-4.503h2.802v.296h-2.496v1.775h2.233v.29h-2.233v1.846h2.586ZM124.408 54.267v4.503h-.252l-2.855-3.912v3.912h-.306v-4.503h.258l2.849 3.912v-3.912h.306ZM125.849 54.267h.305v4.503h-.305v-4.503ZM127.563 56.859v-2.592h.304v2.58c0 1.125.504 1.652 1.377 1.652.874 0 1.372-.527 1.372-1.653v-2.579h.304v2.592c0 1.286-.64 1.942-1.683 1.942-1.042 0-1.675-.656-1.675-1.942h.001ZM136.296 58.77l-.006-3.853-1.754 3.255h-.149l-1.754-3.243v3.841h-.299v-4.503h.257l1.874 3.48 1.873-3.48h.252l.007 4.503h-.301ZM139.845 54.563v1.923h2.234v.296h-2.234v1.988h-.305v-4.503h2.802v.296h-2.497ZM142.937 56.52c0-1.307.927-2.285 2.179-2.285 1.252 0 2.179.971 2.179 2.284 0 1.314-.934 2.283-2.179 2.283-1.245 0-2.179-.978-2.179-2.283Zm4.047 0c0-1.14-.803-1.983-1.868-1.983-1.066 0-1.874.843-1.874 1.982 0 1.14.803 1.981 1.874 1.981 1.071 0 1.868-.841 1.868-1.98ZM151.312 58.77l-1-1.505c-.132.02-.263.032-.406.032h-1.197v1.473h-.306v-4.503h1.503c1.054 0 1.688.566 1.688 1.518 0 .713-.36 1.209-.981 1.409l1.047 1.576h-.348Zm-1.406-1.763c.897 0 1.377-.45 1.377-1.22 0-.771-.48-1.224-1.377-1.224h-1.197v2.444h1.197ZM157.089 58.474v.296h-2.892v-4.503H157v.296h-2.497v1.775h2.234v.29h-2.234v1.846h2.586ZM161.007 58.77l-1.443-2.084-1.436 2.084h-.359l1.61-2.322-1.496-2.18h.358l1.336 1.935 1.334-1.936h.342l-1.497 2.174 1.611 2.33h-.36ZM161.735 56.52c0-1.314.928-2.285 2.174-2.285.592 0 1.125.206 1.491.625l-.192.211c-.359-.38-.797-.534-1.288-.534-1.072 0-1.88.85-1.88 1.982 0 1.133.808 1.981 1.88 1.981.491 0 .928-.16 1.288-.54l.192.213c-.366.418-.898.63-1.491.63-1.245 0-2.174-.97-2.174-2.282v-.002ZM169.297 58.474v.296h-2.892v-4.503h2.803v.296h-2.497v1.775h2.233v.29h-2.233v1.846h2.586ZM170.307 54.267h.306v4.207h2.413v.296h-2.719v-4.503ZM173.856 54.267h.306v4.207h2.413v.296h-2.719v-4.503ZM180.298 58.474v.296h-2.892v-4.503h2.802v.296h-2.496v1.775h2.233v.29h-2.233v1.846h2.586ZM184.72 54.267v4.503h-.252l-2.855-3.912v3.912h-.305v-4.503h.257l2.849 3.912v-3.912h.306ZM185.826 56.52c0-1.314.927-2.285 2.173-2.285.592 0 1.126.206 1.491.625l-.192.211c-.358-.38-.797-.534-1.287-.534-1.073 0-1.881.85-1.881 1.982 0 1.133.808 1.981 1.881 1.981.49 0 .927-.16 1.287-.54l.192.213c-.365.418-.897.63-1.491.63-1.244 0-2.173-.97-2.173-2.282v-.002ZM193.388 58.474v.296h-2.892v-4.503h2.802v.296h-2.496v1.775h2.233v.29h-2.233v1.846h2.586ZM98.84 24.643c4.086 2.083 6.651 6.252 6.651 11.34 0 7.552-6.064 13.272-14.07 13.272H71.43V2.645h19.59c7.691 0 13.303 5.043 13.303 11.93 0 4.164-2.171 7.91-5.484 10.068Zm-5.859-7.942c0-2.419-1.542-4.057-4.041-4.057H82.9v8.115h6.039c2.5 0 4.041-1.638 4.041-4.058Zm.69 17.755c0-2.475-1.691-4.22-4.279-4.22h-6.491v8.477h6.491c2.588 0 4.279-1.743 4.279-4.258v.001ZM110.989 2.644h12.107v46.61h-12.107V2.644ZM160.506 41.3h-3.723v7.955h-10.588v-7.956h-19.78v-2.166l13.2-34.158h10.942l-10.205 26.401h6.589l2.417-9.751h7.424v9.751h3.723v9.922l.001.001ZM193.388 38.116v11.138h-28.277V2.644h27.752v11.1h-15.855v6.861h14.245v10.532h-14.245v6.978h16.38v.001Z"
        fill="#fff"
      />
    </svg>
  )
}

export { BigLogo, SmallLogo }
