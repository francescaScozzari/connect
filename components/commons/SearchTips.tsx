import React from 'react'
import { styled } from 'styled-components'

import { Text } from '@/components/commons/Typography'
import { IconNote, IconTarget } from './Icons'

type Props = {
  titleColor?: string
  textColor?: string
}

const SearchTips = ({
  titleColor = '#2B2D42',
  textColor = '#242424'
}: Props) => {
  return (
    <Wrapper>
      <SubWrapper>
        <IconNote title={'note'} color={titleColor} />
        <Paragraph color={textColor}>
          <Title color={titleColor}>Start from the research call</Title>
          <br /> If you are working or will have to work on a research paper for
          a call, you might report in the search field the title or objectives
          of that
        </Paragraph>
      </SubWrapper>

      <SubWrapper>
        <IconTarget title={'target'} color={titleColor} />
        <Paragraph color={textColor}>
          <Title color={titleColor}>More detail for greater match</Title>
          <br /> Enter as much information, don't limit yourself to a single
          word, that way we can narrow it down and suggest the researchers best
          suited to your need
        </Paragraph>
      </SubWrapper>
    </Wrapper>
  )
}

const Wrapper = styled.div`
  display: flex;
  gap: 2em;
  max-width: 67em;
  margin: 4em 0;

  @media (max-width: 1280px) {
    max-width: 46em;
  }

  @media (max-width: 768px) {
    max-width: 37.5em;
  }
`

const SubWrapper = styled.div`
  display: flex;
  gap: 0.5em;
  padding: 1.5em;
  align-items: start;
  max-width: 29.85em;
  border: 1px solid #7a83ab;
  border-radius: 1.25em;

  svg {
    flex-shrink: 0;
  }
`

const Paragraph = styled(Text.Normal)<{ color: string }>`
  line-height: 1.5;
  color: ${({ color }) => color};
`

const Title = styled.span<{ color: string }>`
  display: inline-block;
  color: ${({ color }) => color};
  margin-bottom: 0.5em;
  font-weight: 600;
  line-height: 1.5;
`

export { SearchTips }
