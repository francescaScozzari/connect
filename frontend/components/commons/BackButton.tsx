import React from 'react'
import styled from 'styled-components'

import { IconSearch } from './Icons'
import { Text } from './Typography'

export const BackButton = () => {
  return (
    <Button>
      <IconSearch title="back" />
      <Label as="span">make a new search</Label>
    </Button>
  )
}

const Label = styled(Text.Normal)`
  text-transform: uppercase;
  font-weight: 600;
  line-height: normal;
  color: white;
`

const Button = styled.button`
  all: unset;
  display: flex;
  justify-content: center;
  align-content: center;
  border-radius: 10px;
  cursor: pointer;
  width: fit-content;
  outline: none;
  padding: 0.75em 1em;
  gap: 1em;
  transition: 0.3s background-color ease-in-out;
  background-color: ${({ theme }) => theme.colors.secondary[0]};

  &:disabled {
    cursor: not-allowed;
    opacity: 0.4;
  }
`
