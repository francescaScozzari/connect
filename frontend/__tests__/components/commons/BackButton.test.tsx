import { expect } from '@jest/globals'
import React from 'react'

import { BackButton } from '@/components/commons/BackButton'
import { renderWithWrappers } from '@/__tests__/functions'

const setup = () => renderWithWrappers(<BackButton />)

describe('Button components', () => {
  describe('<BackButton />', () => {
    test('BackButton renders correctly', () => {
      const container = setup()
      expect(container.firstChild).toMatchSnapshot()
    })
  })
})
