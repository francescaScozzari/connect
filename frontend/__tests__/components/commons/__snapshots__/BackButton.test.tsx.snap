// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button components <BackButton /> BackButton renders correctly 1`] = `
.c1 {
  color: #1C1C1C;
  margin: 0;
  text-transform: none;
}

.c2 {
  font-size: 16px;
  font-weight: 400;
}

.c3 {
  text-transform: uppercase;
  font-weight: 600;
  line-height: normal;
  color: white;
}

.c0 {
  all: unset;
  display: flex;
  justify-content: center;
  align-content: center;
  border-radius: 10px;
  cursor: pointer;
  width: fit-content;
  outline: none;
  padding: 0.75em 1em;
  gap: 1em;
  transition: 0.3s background-color ease-in-out;
  background-color: #CE2F7C;
}

.c0:disabled {
  cursor: not-allowed;
  opacity: 0.4;
}

<button
  class="c0"
>
  <svg
    height="24"
    width="24"
    xmlns="http://www.w3.org/2000/svg"
  >
    <title>
      back
    </title>
    <defs>
      <style />
    </defs>
    <g
      id="_21.search"
    >
      <circle
        cx="9"
        cy="9"
        r="7"
        style="fill: none; stroke: white; stroke-width: 2px;"
      />
      <path
        d="m14 14 8 8"
        style="fill: none; stroke: white; stroke-width: 2px;"
      />
    </g>
  </svg>
  <span
    class="c1 c2 c3"
  >
    make a new search
  </span>
</button>
`;
